src/stability_sdk/client.py

```diff
@@ -284,22 +284,26 @@ def generate(
             elif not isinstance(p, generation.Prompt):
                 raise TypeError("prompt must be a string or generation.Prompt object")
             prompts.append(p)
-
-        step_parameters = dict(
-            scaled_step=0,
-            sampler=generation.SamplerParameters(cfg_scale=cfg_scale),
-            schedule=generation.ScheduleParameters(
-                start=start_schedule,
-                end=end_schedule,
-            ),
-        )
             
         if init_image is not None:
+            step_parameters = dict(
+                scaled_step=0,
```

Comment:
instead of initializeint step_parameters dict two different ways, let's initialize it above this block as before bust just without the schedule key, and only add an entry for schedule after the init_image is not none test